File: lib/py/uapi/internal/generation/GenerateRandomUMockCall.py
from uapi.internal.generation.GenerateContext import GenerateContext
from uapi.internal.generation.GenerateRandomUnion import generate_random_union
from uapi.internal.types.UFn import UFn
from uapi.internal.types.UType import UType


def generate_random_u_mock_call(types: dict[str, UType], ctx: GenerateContext) -> object:
    functions = [
        value for key, value in types.items()
        if isinstance(value, UFn) and not key.endswith('_')
    ]

    functions.sort(key=lambda fn: fn.name)

    selected_fn = functions[ctx.random_generator.next_int_with_ceiling(
        len(functions))]

    return generate_random_union(selected_fn.call.cases, ctx.copy(always_include_required_fields=False))

File: lib/java/src/main/java/uapi/internal/generation/GenerateRandomUMockCall.java
package uapi.internal.generation;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

import uapi.internal.types.UFn;
import uapi.internal.types.UType;

public class GenerateRandomUMockCall {

    public static Object generateRandomUMockCall(Map<String, UType> types, GenerateContext ctx) {
        List<UFn> functions = types.entrySet().stream()
                .filter(entry -> entry.getValue() instanceof UFn)
                .filter(entry -> !entry.getKey().endsWith("_"))
                .map(entry -> (UFn) entry.getValue())
                .sorted((fn1, fn2) -> fn1.getName().compareTo(fn2.getName()))
                .collect(Collectors.toList());

        UFn selectedFn = functions.get(ctx.randomGenerator.nextIntWithCeiling(functions.size()));

        return GenerateRandomUnion.generateRandomUnion(null, false, selectedFn.call.cases,
                ctx.copyWithNewAlwaysIncludeRequiredFields(false));
    }
}
